### ISSUE TITLE
A problem about your contour train

### ISSUE THREAD

**Issue by cccccccccy** [zju3dv/snake#106]
When I read your Evolution Code, I get confused about your contour training design. Can you tell me why the predicted extreme_points are not used for the next contour prediction train? In my opinion, the initial contour in the prediction of the offset from the initial contour to the final contour is not related to the prediction of the previous extreme_point during training. Is my understanding correct？Thank you very much !
`py_pred = self.evolve_poly(self.evolve_gcn, cnn_feature, init['i_it_py'], init['c_it_py'], init['py_ind'])`
in
` 
def forward(self, output, cnn_feature, batch=None):
        #           output = self.gcn(output, cnn_feature, batch)
        #           output.update({'ct': ct, 'detection': detection})
        #           ct（batch，K，2）：（batch, K, [xs,ys])  detection : (torch.cat([bboxes, scores, clses], dim=2) k == 100
        ret = output
        print('cnn_feature.size()', cnn_feature.size()) # cnn_feature.size() == torch.size([1, 64, 128, 96])
        if batch is not None and 'test' not in batch['meta']:
            with torch.no_grad():
                init = self.prepare_training(output, batch)

            ex_pred = self.init_poly(self.init_gcn, cnn_feature, init['i_it_4py'], init['c_it_4py'], init['4py_ind'])
            # 预测极点
            ret.update({'ex_pred': ex_pred, 'i_gt_4py': output['i_gt_4py']})

            # with torch.no_grad():
            #     init = self.prepare_training_evolve(output, batch, init)

            py_pred = self.evolve_poly(self.evolve_gcn, cnn_feature, init['i_it_py'], init['c_it_py'], init['py_ind'])
            # init['i_it_py']
            py_preds = [py_pred]
            for i in range(self.iter):
                py_pred = py_pred / snake_config.ro
                c_py_pred = snake_gcn_utils.img_poly_to_can_poly(py_pred)
                evolve_gcn = self.__getattr__('evolve_gcn'+str(i))
                py_pred = self.evolve_poly(evolve_gcn, cnn_feature, py_pred, c_py_pred, init['py_ind'])
                py_preds.append(py_pred)
            ret.update({'py_pred': py_preds, 'i_gt_py': output['i_gt_py'] * snake_config.ro})`

**Comment by pengsida** [comment]
Yes, the initial contour in the prediction of the offset from the initial contour to the final contour is not related to the prediction of the previous extreme_point
I found that using extreme_points for the next contour prediction training does not improve the performance.

**Comment by cccccccccy** [comment]
Thank you very much!

**Comment by cccccccccy** [comment]
Sorry to bother you again. Since the prediction of extreme_points and contours is irrelevant, is it possible to train 4py and pys at the same time, for example, to make two branches in the head.

**Comment by pengsida** [comment]
The code trains 4py and pys at the same time.
